pallets/xyk/src/lib.rs

```diff
@@ -429,17 +429,29 @@ impl<T: Trait> Module<T> {
         output_reserve: Balance,
         sell_amount: Balance,
     ) -> Result<Balance, DispatchError> {
-        let input_reserve_saturated: U256 = input_reserve.into();
-        let output_reserve_saturated: U256 = output_reserve.into();
-        let sell_amount_saturated: U256 = sell_amount.into();
+        let after_fee_percentage: u128 = 10000 - FEE_PERCENTAGE;
+        let input_reserve_saturated: U256 = input_reserve.saturated_into::<u128>().into();
```

Comment:
Here and later:
use new mechanism from develop for the conversions instead of saturation conversions